src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,360 @@
+    }
+
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
+            @NotNull final List<ActionMenuHolder> parsedHolders,
+            final String potentialArgumentValue)
+    {
+        // TODO: selector support, such as used by CommandKill to find player
+        // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
+
+        switch (this)
+        {
+            case INTEGER:
+                try
+                {
+                    return Integer.parseInt(potentialArgumentValue);
+                }
+                catch (final NumberFormatException e)
+                {
+                    return null;
+                }
+            case BOOLEAN:
+                return Boolean.parseBoolean(potentialArgumentValue);
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                try
+                {
+                    return Integer.parseInt(potentialArgumentValue);
+                }
+                catch (final NumberFormatException e)
+                {
+                    return null;
+                }
+            case PLAYER:
+                final List<String> playerNameStrings = getOnlinePlayerNames(server);
+                if (playerNameStrings.contains(potentialArgumentValue))
+                {
+                    return server.getPlayerList().getPlayerByUsername(potentialArgumentValue);
+                }
+                else
+                {
+                    if ("[abandoned]".equals(potentialArgumentValue))
+                    {
+                        return new FakePlayer(server.getWorld(0), new GameProfile(UUID.randomUUID(), "[abandoned]"));
+                    }
+                    return null;
+                }
+            case COLONY:
+                final List<String> colonyNumberStrings = getColonyIdStrings();
+                try
+                {
+                    int colonyNumber = Integer.parseInt(potentialArgumentValue);
+                    if (sender instanceof EntityPlayer)
+                    {
+                        if (colonyNumber == -1)
+                        {
+                            final IColony icolony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
+                            if (icolony != null)
+                            {
+                                colonyNumber = icolony.getID();
+                            }
+                        }
+                    }
+                    if (colonyNumberStrings.contains(String.valueOf(colonyNumber)))
+                    {
+                        return ColonyManager.getColony(colonyNumber);
+                    }
+                }
+                catch (final NumberFormatException e)
+                {
+                    return null;
+                }
+            case CITIZEN:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 27 to at most 10, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)